src/metrics/_cluster/health.rs

```diff
@@ -17,27 +17,23 @@ async fn metrics(exporter: &Exporter) -> Result<Vec<Metrics>, elasticsearch::Err
         .send()
         .await?;
 
-    let values = response
-        .json::<CluserHealthResponse>()
-        .await?
-        .into_value(inject_cluster_health);
+    let values = response.json::<CluserHealthResponse>().await?.into_value();
 
-    Ok(metric::from_value(values))
-}
+    update_health_metrics_from_value(&values, exporter.metrics().cluster_health_status.clone());
 
-lazy_static! {
-    static ref CLUSTER_STATUS: IntGaugeVec = register_int_gauge_vec!(
-        "elasticsearch_cluster_health_status",
-        "Whether all primary and replica shards are allocated.",
-        &["cluster", "color"]
-    )
-    .expect("valid prometheus metric");
+    Ok(metric::from_value(values))
 }
 
 const COLORS: [&str; 3] = ["red", "green", "yellow"];
 
+fn update_health_metrics_from_value(value: &Value, health_metric: IntGaugeVec) {
+    if let Some(map) = value.as_object() {
+        update_health_metrics(map, health_metric)
+    }
+}
+
 // elasticsearch_cluster_health_cluster_status{cluster="some",status="green"} 1
-fn inject_cluster_health(map: &mut SerdeMap<String, Value>) {
+fn update_health_metrics(map: &SerdeMap<String, Value>, health_metric: IntGaugeVec) {
```

Comment:
not very fond of this rename all other metric sub-systems have `inject_` as a prefix

Reply:
I changed the name because it was misleading here - json map is only used to get values for the metric, it is not being mutated (nothing gets injected). Other `inject_` methods mutate the given value. I personally prefer different name here to explicitly describe that this method only updates the metric value.